agent/utils/dockerapi_compare_versions.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2014-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package utils
+
+import   (
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+type Docker_version string
+
+type dockerapi_version struct {
+	major             int
+	minor             int
+}
+
+// DockerVersionMatches returns whether or not a version matches a given selector.
+// The selector can be any of the following:
+//
+// * x.y -- Matches a version exactly the same as the selector version
+// * >=x.y -- Matches a version greater than or equal to the selector version
+// * >x.y -- Matches a version greater than the selector version
+// * <=x.y -- Matches a version less than or equal to the selector version
+// * <x.y -- Matches a version less than the selector version
+// * x.y,a.b -- Matches if the version matches either of the two selector versions
+func (lhs Docker_version) DockerVersionMatches(selector string) (bool, error) {
```

Comment:
nit: i think the function name can just be "Matches". we know that it's a docker version match from the recevier